src/tox/config/parallel.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import, unicode_literals
+
+from argparse import ArgumentTypeError
+
+ENV_VAR_KEY = "TOX_PARALLEL_ENV"
+OFF_VALUE = 0
+DEFAULT_PARALLEL = OFF_VALUE
+
+
+def auto_detect_cpus():
```

Comment:
some prior art that's worth comparing:

- [seems it was copied from here and has drifted](https://github.com/pytest-dev/pytest-xdist/blob/dd63d0b6674be0372e0e47b452c96912a53f0ad3/xdist/plugin.py#L7-L27)
- [less cases, still the travis workaround](https://github.com/pre-commit/pre-commit/blob/46ae88c1f06b8dc59f3e049c68b2dcea22bb9a54/pre_commit/languages/helpers.py#L59-L70)

Reply:
notably: I can't see a case where `os.cpu_count` exists but `os.sched_getaffinity` does not 🤷‍♂️

Reply:
do we need the Travis? I'm not expecting people running parallel mode on Travis. They probably should prefer parallel Travis job instead, not ?

Reply:
Actually we added that support in `pytest-xdist` because people were running in parallel on Travis. 🤷‍♂️

Reply:
Yeah, there it makes sense to parallelize tests. However in case parallelizing tox targets seems more idiomatic to parallelize at CI level (e.g. separate jobs). This is how tox-travis behaves too.